plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+The DA Server is responsible for verifying that the data returned from the DA layer matches the commitment.
+
+
+### Alternatives
+
+We considered moving the Data Fetching logic into the op-node. It would be possible to place it the fetcher component behind a Go interface; however we rejected this solution in favor of putting the fetching logic over a network boundary.
+Moving the logic to a dedicated HTTP server means that the same op-node binary can work with multiple layers.
+The other option to avoid compiling in the DA Layer specific logic is to use [go plugins](https://pkg.go.dev/plugin). We reject this solution because it requires that the DA fetching logic be implementing in Go and it add places restrictions on how the resulting binary may be used.
+
+## Generic Commitment
+
+The generic commtiment is [specified here](https://github.com/ethereum-optimism/specs/blob/main/specs/experimental/plasma.md#input-commitment-submission).
+The relevant part is copied to this document:
+
+The da-service commitment is as follows: `da_layer_byte ++ payload`. The DA layer byte must be initially restricted to the range `[0, 127)`. This specification will not apportion DA layer bytes, but different DA layers should coordinate to ensure that the DA layer bytes do not conflict. DA Layers can do so in [this discussion](https://github.com/ethereum-optimism/specs/discussions/135). The payload is a bytestring which is up to the DA layer to specify. The DA server should be able to parse the payload, find the data on the DA layer, and verify that the data returned from the DA layer matches what was committed to in the payload.
+
+
+### Alternatives
+
+We considered and rejected making the op-node aware of each DA Layer's commitment format.
+This would allow the op-node to be stricter about validating which DA layers are being used and relies less on the DA Server.
+The downside is that this requires each DA Layer to integrate with the specs and op-node source.
+
+
+## Data Availability Challenge
+
+The Data Availability Challenge ensures that nodes are able to fetch the data corresponding to the commitment posted on L1.
+If the data is not available, nodes should not attempt to use the commitment.
+If the data is only made available to a small portion of nodes, it should either be made available to all nodes by posting the data to L1 or the commitment should be rejected.
+Ensuring that nodes have a consistent view of the data is important to ensure that nodes derive the same chain and the fault proofs function correctly.
+
+Each DA Layer will have to implement logic on L1 to validate their commitments. This logic will emit a specific set of events that the op-node will read.
+These events then get adapted into a golang interface that the derivation pipeline uses to manage L2 reorgs & finality.
+
+### Data Availability Challenge Smart Contract Specification
```

Comment:
Can we describe how this contract will be integrated into the dispute protocol? I see that the current `DataAvailabilityChallenge` contract has not yet been integrated, which should be the case before launch.

We touched on this back in November for the Lattice team, and they seem to have never done it (cc @tynes, who was leading this discussion at the time). The dispute protocol is meant to be generic enough to allow for disputes to exist over varying types of information. If the abstractions aren't suitable for the DA challenge, we need to improve them. 

If we do not bring this contract into the dispute protocol, we risk having several different methods for disputing information, where we could have had standardized interfaces and behavior.

Reply:
So far we haven't be planning on integrating it into the dispute protocol. One of the things about integrating it into the dispute protocol is that the op-node relies on this SC for derivation while it doesn't rely on the dispute protocol for derivation.

Reply:
The main complaint around using the dispute protocol vs the singleton contract is that the dispute protocol requires watching many contracts. It would need to be very easy to find all open disputes (represented by diff contracts) in a stateless way. The challenger prob does this so its likely possible, I had trouble getting buy in from Lattice at the time due to this unknown.

I do think it makes sense to develop tooling around a dispute protocol, especially as we move plasma to be fault provable, we will need to weigh this tech debt against other projects in the future and maybe one day we can unify

Reply:
Two step tracking: watch the factory which emits an a contract address as an event which then emits a status